lib/Common/CommonDefines.h

```diff
@@ -140,7 +140,11 @@
 #define ENABLE_JS_ETW                               // ETW support
 #else
 #define SYSINFO_IMAGE_BASE_AVAILABLE 0
+#ifndef ENABLE_VALGRIND
 #define ENABLE_CONCURRENT_GC 1
+#else
+#define ENABLE_CONCURRENT_GC 0
```

Comment:
Not sure if it was the places you were pointing but it defines to 1 there. see also line 152 -> `#if ENABLE_CONCURRENT_GC`